cmd/specdiff/differ_test.go

```diff
@@ -0,0 +1,282 @@
+package specdiff
+
+import (
+	"io/ioutil"
+	"reflect"
+	"testing"
+
+	"github.com/apigee/registry/rpc"
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+	"github.com/tufin/oasdiff/diff"
+	"google.golang.org/protobuf/testing/protocmp"
+)
+
+type testStruct struct {
+	Name    string            `json:"name,omitempty"`
+	TestMap map[string]string `json:"testmap,omitempty"`
+}
+
+func TestDiffProtoStruct(t *testing.T) {
+	tests := []struct {
+		desc         string
+		baseSpec     string
+		revisionSpec string
+		wantProto    *rpc.Diff
+	}{
+		{
+			desc:         "Struct Diff Added",
+			baseSpec:     "./test-specs/base-test.yaml",
+			revisionSpec: "./test-specs/struct-test-add.yaml",
+			wantProto: &rpc.Diff{
+				Added: []string{
+					"components.schemas.Pet.required.age",
+					"components.schemas.Pet.properties.age",
+				},
+				Deleted:      []string{},
+				Modification: map[string]*rpc.Diff_ValueChange{},
+			},
+		},
+		{
+			desc:         "Struct Diff Deleted",
+			baseSpec:     "./test-specs/base-test.yaml",
+			revisionSpec: "./test-specs/struct-test-delete.yaml",
+			wantProto: &rpc.Diff{
+				Added: []string{},
+				Deleted: []string{
+					"components.schemas.Pet.required.name",
+				},
+				Modification: map[string]*rpc.Diff_ValueChange{},
+			},
+		},
+		{
+			desc:         "Struct Diff Modified",
+			baseSpec:     "./test-specs/base-test.yaml",
+			revisionSpec: "./test-specs/struct-test-modify.yaml",
+			wantProto: &rpc.Diff{
+				Added:   []string{},
+				Deleted: []string{},
+				Modification: map[string]*rpc.Diff_ValueChange{
+					"info.version": {
+						To:   "1.0.1",
+						From: "1.0.0",
+					},
+					"components.schemas.Pet.properties.tag.type": {
+						To:   "integer",
+						From: "string",
+					},
+					"components.schemas.Pet.properties.tag.format": {
+						To: "int64",
+					},
+				},
+			},
+		},
```

Comment:
Consider adding a case to test what happens when you rearrange the ordering of various parts of the spec.

I expect that fields are likely able to appear in any order without the diff changing, but what happens if we change the order of entries in a list of required fields? Requiring `id` and `name` is the same as requiring `name` and `id` so I don't think there should be anything in the diff about it.